lib/expired_record_cleaner.rb

```diff
@@ -0,0 +1,53 @@
+class ExpiredRecordCleaner
+  attr_reader :verbose, :fiscal_year_start
+
+  def initialize(dtim, ok_to_act = false, verbose = false)
+    @fiscal_year_start = self.class.fiscal_year_start(dtim)
+    @ok_to_act = ok_to_act
```

Comment:
it's a safeguard. assumes that by default it will be a no-op.

I typically do named args when I get more than 3 arguments, or if they are all required. That's just a personal preference (you're talking to an old C programmer).